src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -0,0 +1,185 @@
+import org.graylog.events.notifications.EventNotificationContext;
+import org.graylog.events.notifications.EventNotificationModelData;
+import org.graylog.events.notifications.EventNotificationService;
+import org.graylog.events.notifications.PermanentEventNotificationException;
+import org.graylog.events.notifications.TemporaryEventNotificationException;
+import org.graylog.events.processor.EventDefinitionDto;
+import org.graylog2.jackson.TypeReferences;
+import org.graylog2.notifications.Notification;
+import org.graylog2.notifications.NotificationService;
+import org.graylog2.plugin.MessageSummary;
+import org.graylog2.plugin.rest.ValidationResult;
+import org.graylog2.plugin.system.NodeId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+public class SlackEventNotification implements EventNotification {
+
+	public interface Factory extends EventNotification.Factory {
+		@Override
+        SlackEventNotification create();
+	}
+
+
+	private static final Logger LOG = LoggerFactory.getLogger(SlackEventNotification.class);
+
+	private final EventNotificationService notificationCallbackService;
+	private final Engine templateEngine ;
+	private final NotificationService notificationService ;
+	private final ObjectMapper objectMapper ;
+	private final NodeId nodeId ;
+	private final SlackClient slackClient;
+
+	@Inject
+	public SlackEventNotification(EventNotificationService notificationCallbackService,
+								  ObjectMapper objectMapper,
+								  Engine templateEngine,
+								  NotificationService notificationService,
+								  NodeId nodeId, SlackClient slackCLient){
+		this.notificationCallbackService = notificationCallbackService;
+		this.objectMapper = requireNonNull(objectMapper);
+		this.templateEngine = requireNonNull(templateEngine);
+		this.notificationService = requireNonNull(notificationService);
+		this.nodeId = requireNonNull(nodeId);
+		this.slackClient = requireNonNull(slackCLient);
+	}
+
+	/**
+	 *
+	 * @param ctx
+	 * @throws PermanentEventNotificationException is thrown with bad webhook url, authentication error type issues
+	 * @throws TemporaryEventNotificationException is thrown for network or timeout type issues
+	 */
+	@Override
+	public void execute(EventNotificationContext ctx) throws PermanentEventNotificationException,TemporaryEventNotificationException {
+		final SlackEventNotificationConfig config = (SlackEventNotificationConfig) ctx.notificationConfig();
+		ValidationResult  result = config.validate();
+		result.getErrors().entrySet().stream().forEach(e -> LOG.error("Invalid configuration for key [{}] and value [{}]",e.getKey() , e.getValue()));
+
+		if(result.failed()){
+			throw new PermanentEventNotificationException("Please verify your Slack Event Configuration");
+		}
```

Comment:
Config validation happens when a user creates/modifies a Notification from the UI. It should not be possible for invalid config to exist at this point in processing.
```suggestion
```

Reply:
https://github.com/Graylog2/graylog-plugin-integrations/issues/611 - i have created a new issue for this commit suggestion.